### ISSUE TITLE
snapshot listener does not work when using the 'arraycontain' in  where clause

### ISSUE THREAD

**Issue by tunguyenatwork** [FirebaseExtended/flutterfire#1743]
I was trying to create the snapshot listener for the collection.  If I used the 'arrayContain' in the 'where clause' then the snapshot didn't listener on any changes of document in the collection.

```dart
.collection('/items')
         .where('companies', arrayContains: "COMPANY_ABC")
         .snapshot().listen((itemsSnapshot) {}
```

this returns a full set of the documents but it does not listen on the data change.

**Comment by dolanmiu** [comment]
I have the same issue, it works on some fields and not others:


Doesn't work:

```dart
FirebaseAuth.instance.currentUser().asStream())
                  .switchMap((user) => Firestore.instance
                      .collection('/items')
                      .where('staffs', arrayContains: user.uid)
                      .limit(1)
                      .snapshots()
```

Works:

```dart
FirebaseAuth.instance.currentUser().asStream())
                  .switchMap((user) => Firestore.instance
                      .collection('/items')
                      .where('owners', arrayContains: user.uid)
                      .limit(1)
                      .snapshots()
```

Seems like `flutterfire` likes "owners"

**Comment by ljanecek** [comment]
I have the same bug in my app, when i use the instance in the stream with the where conditions, reactivity is not working. Doesn't matter on the type of condition.

```
firebase_core: ^0.4.3+1
firebase_auth: ^0.15.3 
firebase_storage: ^3.1.1
cloud_firestore: ^0.13.0+1
```

**Comment by Salakar** [comment]
Hey all, as part of our on-going work for #2582, this has been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
